### PR TITLE
Add 3D (ZCYX) parsing for MolecularDevices datasets

### LINKED ISSUE #4: MolecularDevices format to zarr: add z plane parsing

### PR DESCRIPTION
**Pull request by imagejan**
Closes #4.

### CODE REVIEW

**Review by tibuch — CHANGES_REQUESTED** [review]
- [ ] add test for `get_well_image_ZCYX`
- [x] expose montage functions
- [ ] add test for stage-montage function
- [ ] add improved test-data

File: src/faim_hcs/io/MolecularDevicesImageXpress.py [hunk 1]
```diff
@@ -134,6 +134,6 @@ def _list_dataset_files(
                 if m_filename:
                     row = m_root.groupdict()
                     row |= m_filename.groupdict()
-                    row["path"] = str(Path(root) / f)
+                    row["path"] = str(Path(root).joinpath(f))
```
Comment (imagejan) [inline src/faim_hcs/io/MolecularDevicesImageXpress.py:137]: Does `joinpath` have any advantages compared to the slash operator? I find the operator more readable and concise.

File: src/faim_hcs/MetaSeriesUtils.py [hunk 2]
```diff
@@ -147,16 +151,85 @@ def sort_key(d):
...
+        plane_files = well_files[well_files["z"] == z]
+        img, ch_hists, ch_metas, meta = get_well_image_CYX(
+            plane_files, assemble_fn=assemble_fn
+        )
+        plane_imgs.append(img)
+        z_positions.append(meta["z-position"])
+        if not channel_histograms:
+            channel_histograms = ch_hists
+        else:
+            [hist1.combine(hist2) for hist1, hist2 in zip(channel_histograms, ch_hists)]
```
[the remark this answers is not in the document]
Comment (imagejan) [reply src/faim_hcs/MetaSeriesUtils.py]: Fixed with https://github.com/fmi-faim/faim-hcs/commit/783ea179c0731b5bc9f5b6a59885940e8e8178a1.

File: tests/test_Zarr.py [hunk 3]
```diff
@@ -231,13 +231,13 @@ def test_write_czyx_image_to_well(self):
             / "E"
             / "7"
             / "0"
-            / "C03_FITC_05_histogram.npz"
+            / "C03_empty_histogram.npz"  # TODO why?
```
Comment (tibuch) [inline tests/test_Zarr.py:234]: If we want to aggregate histograms later on we can assume that all of them exist. Even if they are empty.

File: src/faim_hcs/MetaSeriesUtils.py [hunk 4]
```diff
@@ -279,7 +279,9 @@ def get_well_image_CZYX(
             zyx = build_stack(plane_imgs)
             stacks.append(zyx)
             channel_histograms.append(UIntHistogram(stacks[-1]))
-            channel_metadata.append(ch_metadata)
+            channel_metadata.append(
+                ch_metadata  # FIXME we take he last occurrence here, should be the first one
```
Comment (tibuch) [inline src/faim_hcs/MetaSeriesUtils.py:283]: Why? Is there some metadata which is affected by first vs last?

File: tests/test_Zarr.py (hunk 3 above)
Comment (tibuch) [reply tests/test_Zarr.py:234]: Okay, got it! The test is just wrong! Great catch!

**Comment by imagejan** [comment]
Thanks for your latest fixes, @tibuch.

I squashed the wip commits and force-pushed. This PR looks good to merge now, in my opinion (provided the tests pass on CI as well).

**Comment by tibuch** [comment]
I agree. I am just finalizing the example notebooks and then it is good to go!

**Review by tibuch — APPROVED** [review]

**Pull request merged**